src/content/graphql-js/Tutorial-Authentication.md

```diff
@@ -42,7 +42,9 @@ app.all(
   createHandler({
     schema: schema,
     rootValue: root,
-    graphiql: true,
+    context: (req) => ({
+      ip: req.raw.ip,
```

Comment:
I don't see `req.raw` in Express' documentation; @enisdenjo does graphql-http wrap the request object when passed to context callback?

Reply:
graphql-sse handler is agnostic to the underlying implementation and req.raw is the request of the implementation. [req is actually a graphql-sse request](https://the-guild.dev/graphql/sse/docs/interfaces/handler.Request).

Reply:
So: "yes" :grin: